transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollChannel.java

```diff
@@ -507,14 +507,13 @@ EpollRecvByteAllocatorHandle newEpollHandle(RecvByteBufAllocator.ExtendedHandle
         }
 
         @Override
-        protected void flush0() {
+        protected final void flush0() {
             // Flush immediately only when there's no pending flush.
             // If there's a pending flush operation, event loop will call forceFlush() later,
             // and thus there's no need to call it now.
-            if (isFlagSet(Native.EPOLLOUT)) {
-                return;
+            if (!isFlagSet(Native.EPOLLOUT)) {
```

Comment:
@Scottmitch seems like the code is now basically the same as before here... Consider revert the change of the file ? Seems like the only change that is "new" is the final keyword

Reply:
yes it is the same as before ... just 1 less code and byte code (see below). Maybe the JIT can optimize this ... but simple enough to do it in code.

```java
    public void function1(boolean v) {
        if (!v) {
            return;
        }
        doWork();
    }

    public void function2(boolean v) {
        if (v) {
            doWork();
        }
    }
```

```
  public void function1(boolean);
    Code:
       0: iload_1
       1: ifne          5
       4: return
       5: aload_0
       6: invokevirtual #6                  // Method doWork:()V
       9: return

  public void function2(boolean);
    Code:
       0: iload_1
       1: ifeq          8
       4: aload_0
       5: invokevirtual #6                  // Method doWork:()V
       8: return
```

Reply:
@Scottmitch I guess at the end it really not matters.. So up to you.